lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1988,6 +1991,8 @@ struct AggOneShot : public mlir::RewritePattern {
         for (Operation *subuser : subField.getResult().getUsers()) {
           if (auto aConnect = dyn_cast<StrictConnectOp>(subuser)) {
             if (aConnect.getDest() == subField) {
+              if (subuser->getParentOp() != lhs->getParentOp())
+                return {};
               if (fields.count(subField.getFieldIndex())) // duplicate write
                 return {};
               fields[subField.getFieldIndex()] = aConnect.getSrc();
```

Comment:
Do we need to check the sources are also in the same block?  Maybe not, idea being if it reaches the connect and the connect and LHS are in the same block, it'll reach them both?  I think that's right just writing it out so we're on same page :+1: .

Reply:
In theory the sources should be equal or above lexically (child blocks would be a new scope).  In practice, some things leak.  This is, of course, super weird to have a conditionally declared thing used outside the condition.

Reply:
> Do we need to check the sources are also in the same block?

I think we don't have to since the connection is in the same block as declaration, source values already dominate the connection and declaration.